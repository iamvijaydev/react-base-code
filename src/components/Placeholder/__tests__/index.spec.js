import React from 'react';
import { shallow, mount } from 'enzyme';

import Placeholder from '../index';
import Styled from '../styled';

describe('<Placeholder /> component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<Placeholder message="Hello world!" />);
  });

  afterAll(() => {
    if (wrapper) wrapper.unmount();
  });

  it('should be defined', () => {
    expect(Placeholder).toBeDefined();
  });

  it('snapshot ok', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly renders mesage', () => {
    expect(wrapper.find(Styled.Message).get(0)).not.toBe(null);
  });

  it('renders no actions by default', () => {
    expect(wrapper.find(Styled.ActionBlock).get(0)).toBe(undefined);
  });
});
