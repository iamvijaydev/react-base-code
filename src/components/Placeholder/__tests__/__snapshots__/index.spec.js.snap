// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Placeholder /> component snapshot ok 1`] = `ShallowWrapper {}`;
